frontend/src/lib/k8s/secret.ts

```diff
@@ -8,6 +8,16 @@ export interface KubeSecret extends KubeObjectInterface {
 
 class Secret extends makeKubeObject<KubeSecret>('secret') {
   static apiEndpoint = apiFactoryWithNamespace('', 'v1', 'secrets');
+  static kind = 'Secret';
+
+  static getBaseObject(): KubeSecret {
+    const baseObject = super.getBaseObject() as KubeSecret;
+    return {
+      ...baseObject,
+      kind: Secret.kind,
+      metadata: { ...baseObject.metadata, name: 'base-secret' },
+    };
+  }
```

Comment:
I was hoping we could make things work without having to override this function everywhere (but only in cases where e.g. the objectName is not matching the kind).

Reply:
What I mean is that I think there should be a way for us to have this logic abstract enough in the cluster.ts that it adapts correctly to most of the subclasses. We should definitely not move individual resources' logic into the cluster.ts file.